functions/Backup-DbaServiceMasterKey.ps1

```diff
@@ -0,0 +1,131 @@
+function Backup-DbaServiceMasterKey {
+    <#
+    .SYNOPSIS
+        Backs up specified service master key.
+
+    .DESCRIPTION
+        Backs up specified service master key.
+
+    .PARAMETER SqlInstance
+        The target SQL Server instance or instances.
+
+    .PARAMETER SqlCredential
+        Allows you to login to SQL Server using alternative credentials.
+
+    .PARAMETER Path
+        The directory to export the key. If no path is specified, the default backup directory for the instance will be used.
+
+    .PARAMETER Credential
+        Pass a credential object for the password
+
+    .PARAMETER SecurePassword
+        The password to encrypt the exported key. This must be a SecureString.
+
+    .PARAMETER WhatIf
+        Shows what would happen if the command were to run. No actions are actually performed.
+
+    .PARAMETER Confirm
+        Prompts you for confirmation before executing any changing operations within the command.
+
+    .PARAMETER EnableException
+        By default, when something goes wrong we try to catch it, interpret it and give you a friendly warning message.
+        This avoids overwhelming you with "sea of red" exceptions, but is inconvenient because it basically disables advanced scripting.
+        Using this switch turns this "nice by default" feature off and enables you to catch exceptions with your own try/catch.
+
+    .NOTES
+        Tags: Certificate, Database
+        Author: Chrissy LeMaire (@cl), netnerds.net
+
+        Website: https://dbatools.io
+        Copyright: (c) 2018 by dbatools, licensed under MIT
+        License: MIT https://opensource.org/licenses/MIT
+
+    .EXAMPLE
+        PS C:\> Backup-DbaServiceMasterKey -SqlInstance server1\sql2016
+        ```
+        ComputerName : SERVER1
+        InstanceName : SQL2016
+        SqlInstance  : SERVER1\SQL2016
+        Filename     : E:\MSSQL13.SQL2016\MSSQL\Backup\server1$sql2016-SMK-20170614162311.key
+        Status       : Success
+        ```
+
+        Prompts for export password, then logs into server1\sql2016 with Windows credentials then backs up the service master key to the default backup directory.
+
+    .EXAMPLE
+        PS C:\> Backup-DbaServiceMasterKey -SqlInstance Server1 -Path \\nas\sqlbackups\keys
+
+        Logs into sql2016 with Windows credentials then backs up the service master key to the \\nas\sqlbackups\keys directory.
+
+    #>
+    [CmdletBinding(SupportsShouldProcess, ConfirmImpact = 'Low')]
+    param (
+        [parameter(Mandatory, ValueFromPipeline)]
+        [DbaInstanceParameter[]]$SqlInstance,
+        [PSCredential]$SqlCredential,
+        [PSCredential]$Credential,
```

Comment:
I don't think we need to use just Credential here because of the standard we use for use of that parameter name.

We use this across commands to mean the cred used for remote access to the OS side of a server.

If this is just for encryption of the key I would say something like KeyCredential may be more appropriate...open to other name suggestions but I do think it needs to be something different than just Credential.

Reply:
I was trying to stay consistent with the signature of `Backup-DbaDbMasterKey` since they do such similar things. I do think it's confusing though - I like KeyCredential, makes it much more obvious.

Reply:
Yeah...if @potatoqualitee agrees we might want to update the parameter naming on that backup command just so we don't mix it up. I don't think there is a real reason to take a credential object just for the password, an example can easily be provided that covers the use of that secure string type that is needed.

But we will let @potatoqualitee make final call....

Reply:
I prefer using credentials for passwords because i dont have to look up the syntax, i just have to use it. that said, i like the idea of updating the name! this is a fab PR, and will merge.